Sagas, small guidance improvements (#6573)
Particular/docs.particular.net: nservicebus/sagas/timeouts.md

snippet: saga-multiple-timeouts

## Changing or revoking timeouts

> After a timeout has been requested, it cannot be changed (i.e. rescheduled) or revoked (i.e. deleted, cancelled). Requesting a timeout with the same state again, but with a different duration or timestamp, will not revoke or ignore the original timeout. The original timeout and the subsequent timeout will both be processed by the saga.

Reviewer: ```suggestion
After a timeout has been requested, it cannot be changed (i.e. rescheduled) or revoked (i.e. deleted or cancelled). Requesting a timeout with the same state again, but with a different duration or timestamp, will not revoke or ignore the original timeout. The original timeout and the subsequent timeout will both be processed by the saga.
```

> A timeout is a delayed message and once requested, a timeout message can already be in transit or queued. Once that has happened there is no way to revoke (delete) a timeout as the message cannot be deleted. It is common to perform a state check (e.g. compare to state version or a timestamp) in a timeout handler to see if the timeout is still applicable for processing or to ignore the message.

Reviewer: Given the changes above, this paragraph is now repeating the fact that you cannot revoke a timeout.

Maybe:

```suggestion
When a saga handles a timeout, it may choose to ignore it, depending on how the saga state has changed since the timeout was requested.
```

snippet: saga-multiple-timeouts

> NOTE: As timeouts are just messages sending a multiple timeouts with the same state will result in that timeout being processed multiple times by the saga.

Reviewer: Another idea:

```suggestion
## Changing timeouts

After a timeout has been requested, it cannot be changed. Requesting a timeout with the same state again, but with a different duration, will not change the original timeout. The original timeout and the subsequent timeout will both be processed by the saga.
```

## Revoking timeouts

A timeout that has been scheduled cannot be revoked. This means that when the timeout timestamp has elapsed then this timeout message will be queued and then processed.

> A timeout is a regular message and once requested, a timeout message can already be in transit or queued. Once that has happened there is no way to revoke (delete) a timeout. Sending a timeout will not invalidate/revoke any prior sent timeout. It is common to perform a state check (e.g. comparing a version number, a timestamp or a status field) in a timeout handler to see if the timeout is still applicable for processing. 

Reviewer: In the context of the above suggestion, I don't think anything needs to change in this section. It can remain purely about revoking timeouts, and all it needs to say is that it cannot be done.

```suggestion
A timeout is a regular message and once requested, a timeout message can already be in transit or queued. Once that has happened there is no way to revoke (delete) a timeout. It is common to perform a state check in a timeout handler to see if the timeout is still applicable for processing. 
```
